### PR TITLE
UHM-7617: Exports failing on Lesotho stemTB system

### PR DESCRIPTION
**Pull request by mogoodrich**

### CODE REVIEW

File: src/main/java/org/bahmni/batch/form/ObservationProcessor.java [hunk 1]
```diff
@@ -59,7 +59,14 @@ public List<Obs> process(Map<String,Object> obsRow) throws Exception {
 			allChildObsIds.add((Integer)obsRow.get("obs_id"));
 		}
 
-		List<Obs> obsRows = fetchAllLeafObs(allChildObsIds);
+		List<Obs> obsRows = null;
+
+		if (allChildObsIds != null && allChildObsIds.size() > 0) {
```
Comment (mseaton) [inline src/main/java/org/bahmni/batch/form/ObservationProcessor.java]: allChildObsIds will never be null here.

File: src/main/java/org/bahmni/batch/form/ObservationProcessor.java [hunk 2]
```diff
@@ -59,7 +59,14 @@ public List<Obs> process(Map<String,Object> obsRow) throws Exception {
...
+			obsRows = fetchAllLeafObs(allChildObsIds);
+		}
+		else {
+			return null;
```
Comment (mseaton) [inline src/main/java/org/bahmni/batch/form/ObservationProcessor.java]: I don't love returning null here, before the method below this one runs.

File: src/main/java/org/bahmni/batch/form/ObservationProcessor.java [hunk 3]
```diff
@@ -59,7 +59,14 @@ public List<Obs> process(Map<String,Object> obsRow) throws Exception {
 			allChildObsIds.add((Integer)obsRow.get("obs_id"));
 		}
 
-		List<Obs> obsRows = fetchAllLeafObs(allChildObsIds);
```
Comment (mseaton) [inline src/main/java/org/bahmni/batch/form/ObservationProcessor.java]: Wouldn't it be better if we just change things in `fetchAllLeafObs` to return an empty List if the `allChildObsIds` passed in are null or an empty List?  Then leave the rest of this method alone?

**Review by mseaton — APPROVED** [review]

**Comment by mogoodrich** [comment]
Updated this based on review feedback, also updated version number to 2.4.2

**Pull request merged**